exploration-module/src/main/java/io/edpn/backend/exploration/adapter/web/dto/mapper/RestSystemDtoMapper.java

```diff
@@ -0,0 +1,30 @@
+package io.edpn.backend.exploration.adapter.web.dto.mapper;
+
+import io.edpn.backend.exploration.adapter.web.dto.RestCoordinateDto;
+import io.edpn.backend.exploration.adapter.web.dto.RestSystemDto;
+import io.edpn.backend.exploration.application.domain.Coordinate;
+import io.edpn.backend.exploration.application.domain.System;
+import io.edpn.backend.exploration.application.dto.CoordinateDto;
+import io.edpn.backend.exploration.application.dto.SystemDto;
+import io.edpn.backend.exploration.application.dto.mapper.SystemDtoMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Slf4j
+public class RestSystemDtoMapper implements SystemDtoMapper {
+
+    public SystemDto map(System system) {
+        return new RestSystemDto(system.name(), coordinateFromSystem(system), system.eliteId(), system.starClass());
+    }
+
+    private CoordinateDto coordinateFromSystem(System system) {
+        if (Optional.ofNullable(system.coordinate()).map(Coordinate::x).isEmpty()) {
+            return null;
```

Comment:
I'm going to be flamed for this, but this is a misuse of `Optional` in my opinion.

This could be a rather simple `if` construct, possibly a ternary. It could even be `Objects.requireNonNull()`.

Reply:
Also, the `isEmpty()` if-block could be `orElse(null)` instead.

But is `null` really something we'd want to return here?